services/batch_manager.js

```diff
@@ -17,14 +17,17 @@ const sequelize = new Sequelize(dbConfig.database, dbConfig.username, dbConfig.p
 });
 
 exports.pre_process_params = async (req, resp) => {
-    if(req.query.type == "admin"){
+    if(req.query.type == "admin" && req.query.sports_id == null){
     const results = await models.Batch.findAll()
     return results
-    }else{
+    }else if (req.query.type == null && req.query.sports_id != null){
+        const results = await models.Batch.findAll({where:{status:"active",sports_id:req.query.sports_id}})
+        return results
+    }
+    else if (req.query.type == null && req.query.sports_id == null){
```

Comment:
we can remove req.query.type == null since it is in both condition.

Reply:
ok will resolve this